patches/0013-p256-armv8-optimizations

```diff
@@ -2929,3 +2318,40 @@ Index: aws-lc/third_party/boringssl/crypto/fipsmodule/ec/asm/p256_beeu-armv8-asm
 +    print $_,"\n";
 +}
 +close STDOUT or die "error closing STDOUT: $!"; # enforce flush
+Index: aws-lc/third_party/boringssl/crypto/fipsmodule/ec/p256-x86_64.c
+===================================================================
+--- aws-lc.orig/third_party/boringssl/crypto/fipsmodule/ec/p256-x86_64.c
++++ aws-lc/third_party/boringssl/crypto/fipsmodule/ec/p256-x86_64.c
+@@ -33,8 +33,8 @@
+ #include "internal.h"
+ #include "p256-x86_64.h"
+ 
+-
+-#if !defined(OPENSSL_NO_ASM) && defined(OPENSSL_X86_64) && \
++#if !defined(OPENSSL_NO_ASM) &&  \
++    (defined(OPENSSL_X86_64) || defined(OPENSSL_AARCH64)) &&    \
+     !defined(OPENSSL_SMALL)
+ 
+ typedef P256_POINT_AFFINE PRECOMP256_ROW[64];
+@@ -555,10 +555,12 @@ static void ecp_nistz256_inv0_mod_ord(co
+ static int ecp_nistz256_scalar_to_montgomery_inv_vartime(const EC_GROUP *group,
+                                                  EC_SCALAR *out,
+                                                  const EC_SCALAR *in) {
++#if defined(OPENSSL_X86_64)
+   if ((OPENSSL_ia32cap_get()[1] & (1 << 28)) == 0) {
+     // No AVX support; fallback to generic code.
+     return ec_simple_scalar_to_montgomery_inv_vartime(group, out, in);
+   }
++#endif
```

Comment:
Why only `x86_64` has this fallback?

Reply:
Because `OPENSSL_ia32cap_get()` is only defined for x86_64. I understand as per the comment, it tests for AVX support, which is needed to call the function `beeu_mod_inverse_vartime`.